api.py

```diff
@@ -0,0 +1,116 @@
+        video.download("./videos/")
+        # Wait until video is downloaded
+        while videoToPlay not in os.listdir("./videos"):
+            sleep(0.5)
+    # Stop interlude
+    interlude_process = process_dict.pop(State.INTERLUDE)
+    interlude_process.terminate()
+    interlude_process.wait()
+    # Start streaming video
+    process_dict[State.PLAYING] = Popen(['ffmpeg', '-re', '-i', f"./videos/{videoToPlay}", 
+                                      '-vf', 'scale=426:240', '-c:v', 'libx264', 
+                                      '-preset', 'veryfast', '-tune', 'zerolatency', 
+                                      '-c:a', 'aac', '-ar', '44100', '-f', 'flv', 
+                                      'rtmp://localhost:1935/live/mystream'],  
+                                      stdout=DEVNULL, stdin=DEVNULL, stderr=DEVNULL)
+    process_dict[State.PLAYING].wait()
+    process_dict.pop(State.PLAYING)
+    # Once video is finished playing (or stopped early), restart interlude
+    process_dict[State.INTERLUDE] = Popen(interlude_command,  stdout=DEVNULL, stdin=DEVNULL, stderr=DEVNULL)
+
+@app.get("/")
+async def root():
+    return {"message": "Hello World"}
+
+@app.get("/state/")
+async def state():
+    if State.PLAYING in process_dict:
+        return { "state": State.PLAYING }
+    else:
+        return { "state": State.INTERLUDE }
+
+@app.get("/current/")
+async def current():
+    return { "title": current_video_data["title"], "thumbnail": current_video_data["thumbnail"] }
+
+@app.post("/play/")
+async def play(url: str):
+    # Check if video is already playing
+    if State.INTERLUDE in process_dict:
+        # Start thread to download video, stream it, and provide a response
+        try:
+            video = YouTube(url)
+            # Check for age restriction exception
+            video.bypass_age_gate()
+            threading.Thread(target=download_and_play, args=(url,)).start()
+            current_video_data["title"] = video.title
+            current_video_data["thumbnail"] = video.thumbnail_url 
+            return { "detail": "Success" }
+        # If download is unsuccessful, give response and reason
+        except AgeRestrictedError:
+            return { "detail": "This video is age restricted :(" }
+        except VideoUnavailable:
+            return { "detail": "This video is unavailable :(" }
+        except RegexMatchError:
+            return { "detail": "That's not a YouTube link buddy ..." }
+        except Exception as e:
+            return { "detail": str(e) }
+    return { "detail": "please wait for the current video to end, then make the request" }
+    
+@app.post("/stop/")
+async def stop():
+    # Check if there is a video playing to stop
+    if State.PLAYING in process_dict:
+        # Stop the video playing subprocess
+        process_dict[State.PLAYING].terminate()
+        return { "detail": "EXITED" }
+    return { "detail": "no video is playing at the moment" }
```

Comment:
lets just return a 200 code with an empty response no matter if we exited or not